### PR TITLE
Gui iga

### PR DESCRIPTION
**Pull request by igasm**
board with view of ships added

### CODE REVIEW

File: client/src/main/java/battleships/App.java [hunk 1]
```diff
@@ -1,28 +1,83 @@
...
     }
 
 
     public static void main(String[] args) {
         launch(args);
     }
+
+    public void showLoginWindow(){
+        try {
+            // Load login window form fxml file
```
Comment (szczepanskikrs) [inline client/src/main/java/battleships/App.java]: JavaDocs over the comments

File: client/src/main/java/battleships/App.java [hunk 2]
```diff
@@ -1,28 +1,83 @@
...
+            FXMLLoader loader = new FXMLLoader();
+            loader.setLocation(App.class.getResource(LOGIN_FXML));
+
+            AnchorPane initLayout = (AnchorPane) loader.load();
+
+            // Show the scene containing the init layout.
```
Comment (szczepanskikrs) [inline client/src/main/java/battleships/App.java]: JavaDocs over the comments

File: client/src/main/java/battleships/App.java [hunk 3]
```diff
@@ -1,28 +1,83 @@
...
+            Scene scene = new Scene(initLayout);
+            primaryStage.setScene(scene);
+
+            // Give the logging access to the main app.
```
Comment (szczepanskikrs) [inline client/src/main/java/battleships/App.java]: JavaDocs over the comments

File: client/src/main/java/battleships/game/Board.java [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+        List<Field> emptyFields = IntStream.range(0, 100).mapToObj(p -> new Field(p)).collect(Collectors.toList());
+        return new Board(emptyFields, map);
+    }
+
+    /**
+     * created for temporary use
+     */
+    void generateExample(){
+        //empty list
+        this.fields = IntStream.range(0, 100).mapToObj(p -> new Field(p)).collect(Collectors.toList());
```
Comment (szczepanskikrs) [inline client/src/main/java/battleships/game/Board.java]: Magic numbers

File: client/src/main/java/battleships/game/Board.java [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+        //add 4-mast ship
+        fields.stream()
+                .filter(p -> (p.getPosition() > 2) && (p.getPosition() < 7))
```
Comment (szczepanskikrs) [inline client/src/main/java/battleships/game/Board.java]: Magic numbers :dango:

File: client/src/main/java/battleships/game/Board.java [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+        //add 1-mast ship x 4
+        fields.get(40).setShipPartOn();
+        fields.get(18).setShipPartOn();
+        fields.get(52).setShipPartOn();
+        fields.get(82).setShipPartOn();
+    }
+
+    StackPane rectangleForPosition(int position){
+        Rectangle rec = new Rectangle();
+        rec.setWidth(30);
```
Comment (szczepanskikrs) [inline client/src/main/java/battleships/game/Board.java]: Magic number replace with DEFAULT_RECTANGLE_WIDTH for example

File: client/src/main/java/battleships/game/Board.java [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+        rec.setHeight(30);
```
Comment (szczepanskikrs) [inline client/src/main/java/battleships/game/Board.java]: Magic number replace with DEFAULT_RECTANGLE_HEIGHT for example

File: client/src/main/java/battleships/game/Board.java [hunk 8]
```diff
@@ -0,0 +1,88 @@
...
+        rec.setFill(colorMap.get(fields.get(position).getState()));
+        rec.setStroke(Color.BLACK);
+        StackPane stackPane = new StackPane();
+        if(fields.get(position).isShoted()){
+            Text text = new Text("x");
```
Comment (szczepanskikrs) [inline client/src/main/java/battleships/game/Board.java]: Magic string :dagger:

File: client/src/main/java/battleships/game/Board.java [hunk 9]
```diff
@@ -0,0 +1,88 @@
...
+        this.colorMap = colorMap;
+    }
+
+    static Board build(){
+        Map<FieldState, Color> map = new HashMap<>();
+        map.put(FieldState.EMPTY, Color.WHITE);
+        map.put(FieldState.UNBROKEN_SHIP_PART, Color.GREEN);
+        map.put(FieldState.BROKEN_SHIP_PART, Color.YELLOW);
+        map.put(FieldState.SUNK_SHIP_PART, Color.RED);
+        List<Field> emptyFields = IntStream.range(0, 100).mapToObj(p -> new Field(p)).collect(Collectors.toList());
```
Comment (szczepanskikrs) [inline client/src/main/java/battleships/game/Board.java]: Magic magic magic numberss woooooo o oo o oo o ooo :dagger:

File: client/src/main/java/battleships/App.java (hunk 1 above)
Comment (igasm) [reply client/src/main/java/battleships/App.java]: done

File: client/src/main/java/battleships/App.java (hunk 2 above)
Comment (igasm) [reply client/src/main/java/battleships/App.java]: done

File: client/src/main/java/battleships/App.java (hunk 3 above)
Comment (igasm) [reply client/src/main/java/battleships/App.java]: done

File: client/src/main/java/battleships/game/Board.java (hunk 9 above)
Comment (igasm) [reply client/src/main/java/battleships/game/Board.java]: changed

File: client/src/main/java/battleships/game/Board.java (hunk 4 above)
Comment (igasm) [reply client/src/main/java/battleships/game/Board.java]: changed

File: client/src/main/java/battleships/game/Board.java (hunk 5 above)
Comment (igasm) [reply client/src/main/java/battleships/game/Board.java]: changed

File: client/src/main/java/battleships/game/Board.java (hunk 6 above)
Comment (igasm) [reply client/src/main/java/battleships/game/Board.java]: done

File: client/src/main/java/battleships/game/Board.java (hunk 7 above)
Comment (igasm) [reply client/src/main/java/battleships/game/Board.java]: done

File: client/src/main/java/battleships/game/Board.java (hunk 8 above)
Comment (igasm) [reply client/src/main/java/battleships/game/Board.java]: done

**Comment by igasm** [comment]
changes done as you requested

**Pull request merged**